recipe/meta.yaml

```diff
@@ -41,25 +44,36 @@ build:
   skip: true  # [win]
   skip: true  # [py<35]
   detect_binary_files_with_prefix: true
+  {% if mpi != 'nompi' %}
+  {% set mpi_prefix = "mpi_" + mpi %}
+  {% else %}
+  {% set mpi_prefix = "nompi" %}
+  {% endif %}
 
   # add build string so packages can depend on
   # dagmc or nodagmc variants explicitly:
   # `openmc * nodagmc_*` for openmc without dagmc
   # `openmc * dagmc_*` for openmc with dagmc
+  # mpi or nompi variants explicitly:
+  # `openmc * mpi_mpich_*` for mpich
+  # `openmc * mpi_*` for any mpi
+  # `openmc * nompi_*` for no mpi
 
-  string: {{ dagmc }}_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ build }}
+  string: {{ dagmc }}_{{ mpi_prefix }}_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ build }}
 
 requirements:
   build:
     - cmake
     - make
-    - dagmc 3.2.1  # [dagmc == 'dagmc']
+    - dagmc * {{ mpi_prefix }}_*  # [dagmc == 'dagmc']
```

Comment:
Before we were pinned to a specific version -- is it intentional to not have that pinning anymore?

Reply:
I think we still need version 3.2.1 of DAGMC for OpenMC 0.13.0

There is a check in OpenMC code that check the version of DAGMC is 3.2.1

Reply:
I could not figure out the correct syntax to include the version in addition to the `* {{ mpi_prefix }}_*` (hence all the commits with the message "trying to get naming correct"). If you know the right syntax, do share. Otherwise I will continue to try different things. I couldn't find an example where both the version and mpi were specified like this.